targets/csharp/source/source/Extensions/Models/PlayFabCloudScriptModelsExtensions.cs.ejs

```diff
@@ -17,55 +24,222 @@ namespace PlayFab.CloudScriptModels
+                return new FunctionContext<TFunctionArgument>()
                 {
                     ApiSettings = settings,
                     CallerEntityProfile = contextInternal.CallerEntityProfile,
                     FunctionArgument = contextInternal.FunctionArgument,
-                    AuthenticationContext = authContext
+                    AuthenticationContext = authContext,
+                    CurrentPlayerId = contextInternal.CallerEntityProfile.Lineage.TitlePlayerAccountId
                 };
             }
         }
 
-        private class FunctionExecutionContextInternal
+        private class FunctionContextInternal
         {
             public TitleAuthenticationContext TitleAuthenticationContext { get; set; }
 
             public ProfilesModels.EntityProfileBody CallerEntityProfile { get; set; }
 
             public TFunctionArgument FunctionArgument { get; set; }
         }
+    }
+
+    public class FunctionContext : FunctionContext<object>
+    {
+    }
+
+    /// <summary>
+    /// Used for extracting the data fields incoming as payload to Azure Functions into their respective objects. 
+    /// Meant to be used with Player PlayStream based Azure Functions
+    /// Provides  fields through ApiSettings and AuthenticationContext, the profile of the caller, as well as the 
+    /// arguments passed to the execution request.
+    /// </summary>
+    /// <typeparam name="TFunctionArgument">Type of FunctionArgument</typeparam>
+    public class FunctionPlayerPlayStreamContext<TFunctionArgument>
+    {
+        public PlayFabApiSettings ApiSettings { get; private set; }
+
+        public PlayFabAuthenticationContext AuthenticationContext { get; set; }
+
+        public PlayStreamEventEnvelope PlayStreamEventEnvelope { get; set; }
+
+        public ServerModels.PlayerProfile PlayerProfile { get; set; }
+
+        public TFunctionArgument FunctionArgument { get; set; }
+
+        public string CurrentPlayerId { get; set; }
 
-        private class TitleAuthenticationContext
+        protected FunctionPlayerPlayStreamContext() { }
+
+        /// <summary>
+        /// Creates a new <c>FunctionPlayerPlayStreamContext</c> out of the incoming request to an Azure Function.
+        /// </summary>
+        /// <param name="request">The request incoming to an Azure Function from the PlayFab server</param>
+        /// <returns>A new populated <c>FunctionPlayerPlayStreamContext</c> instance</returns>
+        public static async Task<FunctionPlayerPlayStreamContext<TFunctionArgument>> Create(HttpRequestMessage request)
         {
-            public string Id { get; set; }
-            public string EntityToken { get; set; }
+            using (var content = request.Content)
+            {
+                var body = await content.ReadAsStringAsync();
+                var contextInternal = PlayFabSimpleJson.DeserializeObject<FunctionPlayerPlayStreamContextInternal>(body);
```

Comment:
You are hard coding yourself to a fixed json serializer, and that's not what you should be doing.

                var json = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
                var contextInternal = json.DeserializeObject<FunctionPlayerPlayStreamContextInternal>(body);